cmd/admin/cert.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"slices"
+	"sync"
+
+	"golang.org/x/crypto/ocsp"
+	"golang.org/x/exp/maps"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/db"
+	berrors "github.com/letsencrypt/boulder/errors"
+	"github.com/letsencrypt/boulder/privatekey"
+	rapb "github.com/letsencrypt/boulder/ra/proto"
+	"github.com/letsencrypt/boulder/revocation"
+	sapb "github.com/letsencrypt/boulder/sa/proto"
+)
+
+// subcommandRevokeCert encapsulates the "admin revoke-cert" command. It accepts
+// many flags specifying different ways a to-be-revoked certificate can be
+// identified. It then gathers the serial numbers of all identified certs, spins
+// up a worker pool, and revokes all of those serials individually.
+//
+// Note that some batch methods (such as -incident-table and -serials-file) can
+// result in high memory usage, as this subcommand will gather every serial in
+// memory before beginning to revoke any of them. This trades local memory usage
+// for shorter database and gRPC query times, so that we don't need massive
+// timeouts when collecting serials to revoke.
+func (a *admin) subcommandRevokeCert(ctx context.Context, args []string) error {
+	subflags := flag.NewFlagSet("revoke-cert", flag.ExitOnError)
+
+	// General flags relevant to all certificate input methods.
+	parallelism := subflags.Uint("parallelism", 10, "Number of concurrent workers to use while revoking certs")
+	reason := subflags.Uint("reason", 0, "Revocation reason (must be 0, 1, 4, 5, or 9)")
```

Comment:
```suggestion
	reason := subflags.Uint("reason", 0, "Revocation reason (must be 0, 1, 4, 5, or 9). The default is 0 (unspecified).")
```

Reply:
The flag package appends "(default: 0)" automatically.